test/libsolidity/util/BytesUtils.cpp

```diff
@@ -159,10 +159,25 @@ string BytesUtils::formatBoolean(bytes const& _bytes)
 
 string BytesUtils::formatHex(bytes const& _bytes)
 {
+	stringstream os;
+
 	soltestAssert(!_bytes.empty() && _bytes.size() <= 32, "");
+
 	u256 value = fromBigEndian<u256>(_bytes);
+	string hexStr = toCompactHexWithPrefix(value);
+
+	// The bytes are left aligned if the hex string has 64 chars (32 bytes) + 2 prefix chars (0x)
```

Comment:
This might not be the right fix. I think the main part where information is lost is in `fromBigEndian<u256>(_bytes);` - because after that line, the length of `_bytes` is not used anymore. So leading zeros are ignored and I think that should be fixed.

Reply:
I agree, unfortunately it's not the right fix. We already have all the information we need at this point: every expected return parameter has an internal type representation called `ABIType`, which has a flag named `alignDeclared`. This flag is being set while parsing and can be read e.g. in `BytesUtils::formatBytes` in order to prefix the output correctly.